cmd/heartbeat/heartbeat.go

```diff
@@ -193,42 +193,60 @@ func buildHeartbeats(params paramscmd.Params) []heartbeat.Heartbeat {
 }
 
 func initHandleOptions(params paramscmd.Params) []heartbeat.HandleOption {
-	return []heartbeat.HandleOption{
-		heartbeat.WithFormatting(heartbeat.FormatConfig{
-			RemoteAddressPattern: remote.RemoteAddressRegex,
-		}),
+	var opts []heartbeat.HandleOption
+
+	if !params.Heartbeat.Filter.AllowUnsavedEntity {
+		opts = append(opts,
+			heartbeat.WithFormatting(heartbeat.FormatConfig{
+				RemoteAddressPattern: remote.RemoteAddressRegex,
+			}),
+		)
+	}
+
+	opts = append(opts,
 		filter.WithFiltering(filter.Config{
+			AllowUnsavedEntity:         params.Heartbeat.Filter.AllowUnsavedEntity,
 			Exclude:                    params.Heartbeat.Filter.Exclude,
 			Include:                    params.Heartbeat.Filter.Include,
 			IncludeOnlyWithProjectFile: params.Heartbeat.Filter.IncludeOnlyWithProjectFile,
 			RemoteAddressPattern:       remote.RemoteAddressRegex,
 		}),
 		heartbeat.WithEntityModifer(),
-		remote.WithDetection(),
-		filestats.WithDetection(filestats.Config{
-			LinesInFile: params.Heartbeat.LinesInFile,
-		}),
-		language.WithDetection(),
-		deps.WithDetection(deps.Config{
-			FilePatterns: params.Heartbeat.Sanitize.HideFileNames,
-		}),
-		project.WithDetection(project.Config{
-			ShouldObfuscateProject: heartbeat.ShouldSanitize(
-				params.Heartbeat.Entity, params.Heartbeat.Sanitize.HideProjectNames),
-			MapPatterns:       params.Heartbeat.Project.MapPatterns,
-			SubmodulePatterns: params.Heartbeat.Project.DisableSubmodule,
-		}),
-		project.WithFiltering(project.FilterConfig{
-			ExcludeUnknownProject: params.Heartbeat.Filter.ExcludeUnknownProject,
-		}),
+	)
+
+	if !params.Heartbeat.Filter.AllowUnsavedEntity {
+		opts = append(opts,
+			remote.WithDetection(),
+			filestats.WithDetection(filestats.Config{
+				LinesInFile: params.Heartbeat.LinesInFile,
+			}),
+			language.WithDetection(),
+			deps.WithDetection(deps.Config{
+				FilePatterns: params.Heartbeat.Sanitize.HideFileNames,
+			}),
+			project.WithDetection(project.Config{
```

Comment:
Skipping project detection means the value from `--project` and `--alternate-project` aren't used?

Reply:
Language and project detection can run under circunstances. I'm adding this validation to this PR.